### PR TITLE
Jay/c 2702 definedocument state management for

### PR DESCRIPTION
**Pull request by RoyalSix**
## Description

[Pull Request Description Here]

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-inbox/src/actions/message-count.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const QUERY_MESSAGE_COUNT = `
+  query MessageCount($isRead: Boolean!) {
+    messageCount(params: {
+      isRead: $isRead
+    })
+  }
+`;
+
+export const getMessageCount = async (client, { isRead = false }) => {
```
Comment (rileylnapier) [inline packages/react-inbox/src/actions/message-count.ts:9]: do we have to pass client in?

File: packages/react-provider/src/graph-ql/index.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import { useMemo } from "react";
-import { createClient } from "urql";
+import { Client, createClient } from "urql";
 
-export default ({ clientKey, userId, userSignature, apiUrl }) => {
-  const client = useMemo(() => {
-    return createClient({
+export class GraphQLClient {
+  client: Client;
+  constructor({ clientKey, userId, userSignature, apiUrl }) {
```
Comment (rileylnapier) [inline packages/react-provider/src/graph-ql/index.ts:5]: what happens if the userId changes?

File: packages/react-provider/src/middleware.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ const asyncMiddleware = (store) => (next) => async (action) => {
   });
 
   try {
-    const result = await action.payload();
+    const result = await action.payload(store.getState());
```
Comment (rileylnapier) [inline packages/react-provider/src/middleware.ts:12]: this feels a little weird

File: packages/storybook/stories/react-inbox/with-toast.stories.tsx [hunk 4]
```diff
@@ -49,6 +50,8 @@ export function Default() {
   return (
     <>
       <CourierProvider
+        apiUrl={API_URL}
+        // wsUrl={WS_URL}
```
Comment (rileylnapier) [inline packages/storybook/stories/react-inbox/with-toast.stories.tsx:54]: do we need wsurl?

File: packages/react-inbox/src/actions/message-count.ts (hunk 1 above)
Comment (RoyalSix) [reply packages/react-inbox/src/actions/message-count.ts:9]: I believe so, I cannot think of another pattern without using hooks, or being on the window object

File: packages/react-provider/src/graph-ql/index.ts (hunk 2 above)
Comment (RoyalSix) [reply packages/react-provider/src/graph-ql/index.ts:5]: If any one these variables change a new client will be created. Check here https://github.com/trycourier/courier-react/blob/d66fbf0e9f59eb5fdc7b8341487fa20729a50863/packages/react-provider/src/index.tsx#L36

File: packages/react-provider/src/middleware.ts (hunk 3 above)
Comment (RoyalSix) [reply packages/react-provider/src/middleware.ts:12]: I updated to use `react-thunk` pattern

File: packages/storybook/stories/react-inbox/with-toast.stories.tsx (hunk 4 above)
Comment (RoyalSix) [reply packages/storybook/stories/react-inbox/with-toast.stories.tsx:54]: In the example, no. I removed

**Review by rileylnapier — APPROVED** [review]
i think we're getting closer.  still a few things that feel off but good to go

**Pull request merged**